relay-metrics/src/aggregation.rs

```diff
@@ -1806,15 +1913,60 @@ impl Aggregator {
                             "Failed to flush the buckets, dropping {batch_size} buckets: {err}."
                         );
                     }
-                });
+                }
+                num_batches += 1;
+                relay_statsd::metric!(
+                    histogram(MetricHistograms::BatchesPerPartition) = num_batches as f64,
+                );
             }
         }
-
         relay_statsd::metric!(histogram(MetricHistograms::BucketsFlushed) = total_bucket_count);
     }
+
+    async fn handle_message(&mut self, msg: Aggregator) {
+        match msg {
+            // This is the `HealthCheck` request where the `AsyncResponse` is expected.
+            Aggregator::AcceptsMetrics(AcceptsMetrics, sender) => {
+                let result = !self
+                    .cost_tracker
+                    .totals_cost_exceeded(self.config.max_total_bucket_bytes);
+                sender.send(result);
+            }
+            #[cfg(test)]
+            Aggregator::BucketCountInquiry(_, sender) => {
+                sender.send(self.buckets.len());
+            }
+            Aggregator::InsertMetrics(InsertMetrics {
+                project_key,
+                metrics,
+            }) => {
+                for metric in metrics {
+                    if let Err(err) = self.insert(project_key, metric) {
+                        relay_log::error!("failed to insert metrics: {}", err)
+                    }
+                }
+            }
+            Aggregator::MergeBuckets(MergeBuckets {
+                project_key,
+                buckets,
+            }) => {
+                if let Err(err) = self.merge_all(project_key, buckets) {
+                    relay_log::error!("failed to merge buckets: {}", err)
```

Comment:
(part 2) These errors were propagated to the sender instead of being logged here. Please restore old behavior -- we can follow up with a dedicated PR that changes control flow.

Reply:
Same as in https://github.com/getsentry/relay/pull/1508#discussion_r990594129 the errors in the sender were never logged or reported - we decided to report them here